wallet-sdk/src/main/kotlin/org/stellar/walletsdk/util/FormatOperation.kt

```diff
@@ -0,0 +1,163 @@
+package org.stellar.walletsdk.util
+
+import org.stellar.sdk.Asset
+import org.stellar.sdk.responses.operations.CreateAccountOperationResponse
+import org.stellar.sdk.responses.operations.OperationResponse
+import org.stellar.sdk.responses.operations.PathPaymentBaseOperationResponse
+import org.stellar.sdk.responses.operations.PaymentOperationResponse
+import org.stellar.walletsdk.*
+
+/**
+ * Format Stellar account operations to make them consistent.
+ *
+ * @param accountAddress Stellar address of the account
+ * @param operation Stellar operation to format
+ *
+ * @return formatted operation
+ */
+fun formatStellarOperation(accountAddress: String, operation: OperationResponse): WalletOperation {
+  when (operation.type) {
+    // Create account
+    StellarOperationTypes.CREATE_ACCOUNT.type -> {
```

Comment:
I think adding extension fields/functions would work better
`OperationResponse.amount: String`
and so on. In that case this function can be removed, and wallet operation is easily created
```kotlin
return WalletOperation(operation.id, operation,.createdAt, operation.amount //extension value, <...>)
```
Example of such function:
```kotlin
fun OperationResponse.amount(): String {
  return when (this) {
    is CreateAccountOperationResponse -> {
      return startingBalance
    }
    // all other branches
    else -> ""
  }
}
```
With this approach it's separation per-field, not per-class

Another option is to add .toWalletOperation() extension on each supported OperationResponse type (and cast manually
```kotlin
when (operation) 
   is CreateAccountOperationResponse -> operation.toWallet() // CreateAccountOperationResponse.toWallet()
   is PaymentOperationResponse -> operation.toWallet()  // PaymentOperationResponse.toWallet() 
   else -> UnsupportedOperationError()
```

Reply:
To me, it feels like this approach would make it harder to reason about logic for a specific type. It's not great that we're setting the same value for 3 of those fields, but it is clearer what is going on for every supported type.

Is there something similar to TypeScript's object spread in Kotlin? In TS, it could be cleaned up like this:

```typescript
const defaults = {
  id: operation.id,
  date: operation.createdAt,
  amount: "",
  account: "",
  asset: [],
  type: WalletOperationType.OTHER,
  rawOperation: operation,
};

const formattedOp = {
  ...defaults,
  amount: operation.amount,
  type: WalletOperationType.SEND,
};
```

Reply:
Unfortunately, there is no object spread operator. Right now I'm not sure if there is a way to make it more cleaner. Another option is to pre-fill some values with defaults, and change them later in switch.
```
val id = operation.id
val date = operation.createdAt
val rawOperation = rawOperation
var type = OTHER
<...>

when (operation) {
   is CreateAccountOperationResponse -> type = if (isCreator) SEND else RECEIVE
   <...>
}

return WalletOperation(id, date, <...>)
```
(You also should type-check OperationResponse vs checking operation.type and casting)